Mathlib/Analysis/SpecialFunctions/ImproperIntegrals.lean

```diff
@@ -51,6 +57,49 @@ theorem integral_exp_neg_Ioi (c : ℝ) : (∫ x : ℝ in Ioi c, exp (-x)) = exp
 theorem integral_exp_neg_Ioi_zero : (∫ x : ℝ in Ioi 0, exp (-x)) = 1 := by
   simpa only [neg_zero, exp_zero] using integral_exp_neg_Ioi 0
 
+theorem integrableOn_exp_mul_complex_Ioi {a : ℂ} (ha : a.re < 0) (c : ℝ) :
+    IntegrableOn (fun x : ℝ => Complex.exp (a * x)) (Ioi c) := by
+  refine (integrable_norm_iff ?_).mp ?_
+  · apply Continuous.aestronglyMeasurable
+    fun_prop
+  · conv in (fun x => _) =>
+      ext x
+      rw [(by simp [Complex.norm_exp] : ‖Complex.exp (a * x)‖ = exp (-(-a.re * x)))]
+    have a_neg : 0 < -a.re := by simp [ha]
+    apply (integrableOn_Ioi_comp_mul_left_iff (fun x => exp (-x)) c a_neg).mpr
+    apply integrableOn_exp_neg_Ioi
+
+theorem integrableOn_exp_mul_complex_Iic {a : ℂ} (ha : 0 < a.re) (c : ℝ) :
+    IntegrableOn (fun x : ℝ => Complex.exp (a * x)) (Iic c) := by
+  conv in (fun x => _) =>
+    ext x
+    rw [(by simp : a * x = -a * (-x : ℝ))]
+  have a_neg : (-a).re < 0 := by simp [ha]
+  exact (integrableOn_exp_mul_complex_Ioi a_neg (-c)).comp_neg_Iic
+
+theorem integral_exp_mul_complex_Ioi {a : ℂ} (ha : a.re < 0) (c : ℝ) :
+    ∫ x : ℝ in Set.Ioi c, Complex.exp (a * x) = - Complex.exp (a * c) / a := by
+  refine
+    tendsto_nhds_unique (intervalIntegral_tendsto_integral_Ioi c
+      (integrableOn_exp_mul_complex_Ioi ha c) tendsto_id) ?_
+  have nonzero : a ≠ 0 := fun h => (Complex.zero_re ▸ h ▸ ha).false
+  simp_rw [integral_exp_mul_complex nonzero, id_eq]
+  rw [(by simp : - Complex.exp (a * c) / a = (0 - Complex.exp (a * c)) / a)]
+  refine Tendsto.div_const (Tendsto.sub_const ?_ _) _
+  apply Complex.tendsto_exp_nhds_zero_iff.mpr
+  simp only [Complex.mul_re, Complex.ofReal_re, Complex.ofReal_im, mul_zero, sub_zero]
+  exact Tendsto.neg_mul_atTop ha tendsto_const_nhds tendsto_id
```

Comment:
1. `aesop` can prove the `have nonzero`, so we can inline it into the `simp_rw`.
2. Your `rw [(by simp...)]` should have just be `rw [← sub_zero]`, but...
3. sometimes it's nicer to just tell Lean the thing you want to prove with `suffices`
```suggestion
  refine tendsto_nhds_unique (intervalIntegral_tendsto_integral_Ioi c
    (integrableOn_exp_mul_complex_Ioi ha c) tendsto_id) ?_
  simp_rw [integral_exp_mul_complex (c := a) (by aesop), id_eq]
  suffices Tendsto (fun x : ℝ ↦ Complex.exp (a * x)) atTop (𝓝 0) by
    simpa using this.sub_const _ |>.div_const _
  simpa [Complex.tendsto_exp_nhds_zero_iff] using tendsto_const_nhds.neg_mul_atTop ha tendsto_id
```